lib/prelude.dx

```diff
@@ -846,6 +834,73 @@ instance Ix Bool
     True -> 1
   unsafe_from_ordinal = \i. i > 0
 
+'## Subset class
+
+interface Subset a_sub a
+  inject' : a_sub -> a
+  project' : a -> Maybe a_sub
+
+def project {a} (a_sub:Type) [Subset a_sub a] (x:a) : Maybe a_sub =
+  project' x
+
+def inject {a_sub} (a:Type) [Subset a_sub a] (x:a_sub) : a =
+  inject' x
+
+instance {a b c} [Subset a b, Subset b c] Subset a c
+  inject' = \x. inject c $ inject b x
+  project' = \x. case project b x of
+    Nothing -> Nothing
+    Just y -> project a y
+
+-- TODO: figure out how to have an instance like this without producing an
+-- infinite number of apparent synthesis paths.
+-- instance {a} Subset a a
+--   inject' = \x. x
+--   inject' = \x. inject c $ inject b x
```

Comment:
remove dead code?

Reply:
Sure. I was keeping it in as a reminder in case someone comes along and naturally tries to implement it. But deleting it sgtm.